pkg/types/userconfig/api.go

```diff
@@ -567,3 +568,149 @@ func ZeroCompute() Compute {
+			event["predictor.models._is_defined"] = true
+			if len(api.Predictor.Models.Paths) > 0 {
+				event["predictor.models.paths._is_defined"] = true
+				event["predictor.models.paths._len"] = len(api.Predictor.Models.Paths)
+				var numSignatureKeysDefined int
+				for _, mmPath := range api.Predictor.Models.Paths {
+					if mmPath.SignatureKey != nil {
+						numSignatureKeysDefined++
+					}
+				}
+				event["predictor.models.paths._num_signature_keys_defined"] = numSignatureKeysDefined
+			}
+			if api.Predictor.Models.Dir != nil {
+				event["predictor.models.dir._is_defined"] = true
+			}
+			if api.Predictor.Models.CacheSize != nil {
+				event["predictor.models.cache_size._is_defined"] = true
+				event["predictor.models.cache_size"] = *api.Predictor.Models.CacheSize
+			}
+			if api.Predictor.Models.DiskCacheSize != nil {
+				event["predictor.models.disk_cache_size._is_defined"] = true
+				event["predictor.models.disk_cache_size"] = *api.Predictor.Models.DiskCacheSize
+			}
+			if api.Predictor.Models.SignatureKey != nil {
+				event["predictor.models.signature_key._is_defined"] = true
+			}
+		}
+
+		if api.Predictor.ServerSideBatching != nil {
+			event["predictor.server_side_batching._is_defined"] = true
+			event["predictor.server_side_batching.max_batch_size"] = api.Predictor.ServerSideBatching.MaxBatchSize
+			event["predictor.server_side_batching.batch_interval"] = api.Predictor.ServerSideBatching.BatchInterval.Seconds()
+		}
+	}
+
+	if api.UpdateStrategy != nil {
+		event["update_strategy._is_defined"] = true
+		event["update_strategy.max_surge"] = api.UpdateStrategy.MaxSurge
+		event["update_strategy.max_unavailable"] = api.UpdateStrategy.MaxUnavailable
+	}
+
+	if api.Autoscaling != nil {
+		event["autoscaling._is_defined"] = true
+		event["autoscaling.min_replicas"] = api.Autoscaling.MinReplicas
+		event["autoscaling.max_replicas"] = api.Autoscaling.MaxReplicas
+		event["autoscaling.init_replicas"] = api.Autoscaling.InitReplicas
+		if api.Autoscaling.TargetReplicaConcurrency != nil {
+			event["autoscaling.target_replica_concurrency._is_defined"] = true
+			event["autoscaling.target_replica_concurrency"] = *api.Autoscaling.TargetReplicaConcurrency
+		}
+		event["autoscaling.max_replica_concurrency"] = api.Autoscaling.MaxReplicaConcurrency
+		event["autoscaling.window"] = api.Autoscaling.Window.Seconds()
+		event["autoscaling.downscale_stabilization_period"] = api.Autoscaling.DownscaleStabilizationPeriod.Seconds()
+		event["autoscaling.upscale_stabilization_period"] = api.Autoscaling.UpscaleStabilizationPeriod.Seconds()
+		event["autoscaling.max_downscale_factor"] = api.Autoscaling.MaxDownscaleFactor
+		event["autoscaling.max_upscale_factor"] = api.Autoscaling.MaxUpscaleFactor
+		event["autoscaling.downscale_tolerance"] = api.Autoscaling.DownscaleTolerance
+		event["autoscaling.upscale_tolerance"] = api.Autoscaling.UpscaleTolerance
+	}
+
+	if api.FileName != "" && api.FileName != "cortex.yaml" {
```

Comment:
Based on the update to the python client, this information may not be useful because an api config yaml file that isn't named `cortex.yaml` will be generated in python.